### ISSUE TITLE
Add CI for hack/local-cluster-up

### ISSUE THREAD

**Issue by cblecker** [kubernetes/test-infra#6696]
It would be great to have a presubmit coverage for changes that touch hack/local-cluster-up.sh (using default settings). If there is a change somewhere in the code base that conflicts with the default settings, this impacts contributors as they need to diagnose what is going on and which settings they need to fix.

Is this a good candidate for DinD? (cc: @BenTheElder)

ref: https://github.com/kubernetes/kubernetes/issues/14207

@kubernetes/sig-contributor-experience-feature-requests

**Comment by BenTheElder** [comment]
Possibly? I'm not sure what coverage we should do exactly, e2es?

**Comment by cblecker** [comment]
I don't even think we need that much.. just like.. run the script through, verify a cluster comes up, and then tear it down.

**Comment by BenTheElder** [comment]
I'm fairly confident this would work fine with the bootstrap image / docker in docker. We could add a run_if_changed job.

**Comment by fejta** [comment]
What does hack/local-up-cluster.sh do? Does someone who uses this regularly want to make a CI job out of it?

**Comment by dims** [comment]
yes @fejta `hack/local-cluster-up.sh` is very handy to test cloud providers. i essentially spin it up in a vm started by openstack and test the functionality. would like to use this in future CI jobs outside of kubernetes CI as well (under https://github.com/theopenlab)

**Comment by dims** [comment]
it essentially runs `hyperkube apiserver` `hyperkube controller-manager` `hyperkube kubelet` etc...

**Comment by fejta** [comment]
would love to see someone provide us with a `kubetest --foo=foo --bar=bar` command that starts up a local cluster. Once we have that I'm sure we can get it into CI

**Comment by cblecker** [comment]
/remove-help
Thanks @dims !

**Comment by dims** [comment]
@cblecker two birds in one stone with #6759 we can be sure that local-up-cluster works! and we can use it to run conformance suite too

**Comment by dims** [comment]
Done! we have both CI and PR jobs. Both running conformance tests. 

https://k8s-testgrid.appspot.com/sig-testing-misc#ci-kubernetes-local-e2e
https://k8s-testgrid.appspot.com/sig-testing-misc#pull-kubernetes-local-e2e

I have 2 PR(s) for the remaining failures:
https://github.com/kubernetes/kubernetes/pull/61000
https://github.com/kubernetes/kubernetes/pull/60811

We can get those in when v1.11 opens up and advertise these jobs and may be make it vote as well
